### PR TITLE
feat(cloudrunci): Add retry by default to Service.Request

### PR DESCRIPTION
**Pull request by muncus**
Functional options pattern allows customizing retry behavior

Also updates several tests to use this new helper.

### CODE REVIEW

File: run/testing/helloworld.e2e_test.go [hunk 1]
```diff
@@ -36,30 +34,17 @@ func TestHelloworldService(t *testing.T) {
 	}
 	defer service.Clean()
 
-	req, err := service.NewRequest("GET", "/")
+	resp, err := service.Request("GET", "/")
+	out, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		t.Fatalf("service.NewRequest: %v", err)
+		t.Errorf("ioutil.ReadAll: %v", err)
```
Comment (telpirion) [inline run/testing/helloworld.e2e_test.go:40]: nit: replace use of `ioutil` with `io`.

**Review by telpirion — APPROVED** [review]
This looks really good to me! I would just make sure that we have usage of this pattern well documented (somewhere).

File: run/testing/helloworld.e2e_test.go (hunk 1 above)
Comment (muncus) [reply run/testing/helloworld.e2e_test.go:40]: is this safe to do while we still (nominally) support versions < 1.18, or should we wait?

Comment (muncus) [reply run/testing/helloworld.e2e_test.go:40]: We discussed this OOB, and decided to wait on updating the package here until 1.18 is our minimum supported version.

**Pull request merged**